GOVERNANCE.md

```diff
@@ -0,0 +1,147 @@
+# CoreDNS Governance
+
+## Principles
+
+The CoreDNS community adheres to the following principles:
+
+- Open: CoreDNS is open source, advertised on [our website](https://coredns.io/community).
+- Welcoming and respectful: See [Code of Conduct](CODE-OF-CONDUCT.md).
+- Transparent and accessible: Work and collaboration are done in public.
```

Comment:
What is the scope of this?  Does it mean that _all_ CoreDNS related discussions should be made known and accessible/open to the public? (e.g. scheduled conference call meetings)

Reply:
Personally I think just like the way cncf works, everything could be public. For example, cncf toc meetings are publicly joinable.

I do understand there might be some special situations. For example, let's say I find one contributor who has done a lot to coredns. I want to send an invite to the contributor to be a maintainer. Sometimes I may want to send an email to `maintainers@coredns.io` first to see if there is any objection, before the invitation email to the contributor is out.

This is for curtesy: you _really_ don't want to send an invitation first, then take it back a couple of days later.

I think there are some special scenarios. Though most of the time I think public is not a problem.

Reply:
Would I be violating the principles if I participate in an ad-hoc whiteboard discussion about CoreDNS with a colleague in person without setting up a public video chat (e.g. Zoom).  I would hope no, as it would stifle in-person collaboration.  But strictly speaking it does violate the principle as written.

Reply:
I think `public` applies only if there is an _impact_ to coredns project.

I believe in person collaborations does not have any impact to coredns project/repo itself, as it does not change coredns repo or organization.

Participation to cncf could change the coredns project. A PR could also change the coredns repo, but of course a PR is always public.